docs/dwarf-op.rst

```diff
@@ -1,158 +1,180 @@
+    specified.
+``all``
+    Selects all dwarves.
+``<name>``
+    Selects any dwarf with <name> anywhere in their name or nickname. This
+    option ignores protection status.
+``named``
+    Selects dwarves with user-given names.
+``unnamed``
+    Selects dwarves without user-given names.
+``employed```
+    Selects dwarves with custom professions. Does not include optimized dwarves.
+``optimized``
+    Selects dwarves that have been previously optimized by ``dwarf-op``.
+``unoptimized``
+    Selects any dwarves that have not been previously optimized by ``dwarf-op``.
+``protected``
+    Selects protected dwarves.
+``unprotected``
+    Selects unprotected dwarves.
+``drunks``
+    Selects any dwarves who have the ``DRUNK`` profession, including those who
+    have been zeroed by the ``--clear`` command option.
+``jobs "[" jobs <jobname> [<jobname> ...] "]"``
+    Selects any dwarves with the specified custom professions.
+``waves "[" waves <num> [<num> ...] "]"``
+    Selects dwarves from the specified migration waves. Waves are enumerated
+    starting at 0 and increasing by 1 with each wave. The waves go by season and
+    year and thus should match what you see in `list-waves` or Dwarf Therapist.
+    It is recommended that you ``--show`` the selected dwarves before modifying
+    them.
+
+Options
+-------
+
+``--reset``
+    Forget which dwarves have been optimized. However, if you reload an existing
+    save, the optimization list will be reloaded.
+``--resetall``
+    Forget which dwarves have been optimized and remove the persisted
+    optimization data.
+``--show``
+    Lists the selected dwarves. Useful for previewing selected dwarves before
+    modifying them or looking up the migration wave number for a group.
+``--clean <value>``
+    Checks for skills with a rating of ``<value>`` and removes them from the
+    dwarf's skill list.
+``--clear``
+    Zeroes the skills and attributes of selected dwarves. No attributes, no
+    labors. Assigns ``DRUNK`` profession.
+``--reroll [inclusive]``
+    Clears attributes of selected dwarves, then rerolls that dwarf based on
+    their jobs. Run ``dwarf-op --list attrib_levels`` to see how stats are
+    distributed. If ``inclusive`` is specified, then attributes are not cleared,
+    but rather will only be changed if the current reroll is better. This
+    command ignores dwarves with jobs that are not listed in the ``jobs`` table.
+``--optimize``
+    Performs a job search for unoptimized dwarves. Run
+    ``dwarf-op --list job_distribution`` to see how jobs are distributed.
+``--applyjobs "[" <job> [<job> ...] "]"``
+    Applies the listed jobs to the selected dwarves. Run
+    ``dwarf-op --list jobs`` to see available jobs.
+``--applyprofessions "[" <profession> [<profession> ...] "]"``
+    Applies the listed professions to the selected dwarves. Run
+    ``dwarf-op --list professions`` to see available professions.
+``--applytypes "[" <profession> [<profession> ...] "]"``
```

Comment:
This syntax, I personally find very confusing. I don't understand what this is trying to communicate. And afaik you definitely don't need to `"[" "]"`

Bottom line: if we're trying to make the docs more accessible to everyone, I think this is too obfuscated for most people to know how to read correctly

Reply:
I feel as though the `[ a b c ]` syntax for lists maybe is the root of this evil. So maybe we could use curly braces instead? I dunno how many other scripts use this, but if it's just this one.. ez pz?

Reply:
it's the "standard" syntax of `argparse.processArgs`, which unfortunately many scripts use. I've offered an alternative with `argparse.stringList` which allows you to represent a list with a comma-separated string, but the script would have to be modified to use it.

It's extra confusing because in the language of usage documentation, square brackets means "optional element" (which is why I'm quoting those square brackets to indicate that they are to be taken literally instead of having their usual meaning)

I'd rather not rewrite scripts just now, though, since I have to focus on getting the documentation updated across all scripts for this release. If you'd like to make (and test) the change though, I'd be happy to merge it!

Reply:
note that the "Examples" section precedes the Options section, and I specifically included examples with these square bracket lists in them to clarify their usage.

Reply:
I'll give it a shot.

Reply:
Ok. Let me merge this PR so you can avoid conflicts later. Feel free to make more doc changes as you change the code!